docs/SoftwareCapabilities.md

```diff
@@ -0,0 +1,96 @@
+The identity of the immediate caller does not matter.
+
+CHERI capabilities to software-defined capabilities
+---------------------------------------------------
+
+CHERI provides a hardware mechanism for building software-defined capabilities: sealing.
+The sealing mechanism allows a pointer (a CHERI capability) to be made immutable and unusable until it is unsealed using an authorising capability.
+The CHERIoT ISA has a limited set of space for sealing types and so these are virtualised with the allocator's [token](../sdk/include/token.h) APIs.
+These APIs combine allocation and sealing, returning both sealed and unsealed capabilities to an object.
+The unsealed capability can be used directly, the sealed capability can be passed to other code and unsealed only by calling the allocator's `token_unseal` function with the capability used to allocate the object.
+
+A compartment can use this to generate software-defined capabilities that represent dynamic resources.
+For example, a network stack can use it to allocate the state associated with a connection.
+The scheduler uses the same mechanism for providing capabilities for cross-thread communication so that, for example, only a holder of the relevant capability can send or receive messages in a message queue.
+
+Static software-defined capabilities
+------------------------------------
+
+In addition to the dynamic software-defined capabilities, it is often useful to provision a set of capabilities to a compartment at build time.
+The simplest case for this is (allocator capabilities)[Allocator.md], which authorise allocation and so are necessary to be able to create any dynamic software-defined capabilities.
+The scheduler also uses this mechanism for capabilities that allow interaction with [interrupts](Interrupts.md).
+
+Objects created in this way are allocated outside of a compartment's global region but are accessed only via capabilities provided by the loader.
+These capabilities use the allocator's sealing mechanism and so can be unsealed only by the owner of the relevant sealing capability.
+
+For more detail on how to use the static sealing mechanism, see [`compartment-macros.h`](../sdk/include/compartment-macros.h).
+
+### Exporting a sealing type
+
+The `STATIC_SEALING_TYPE` macro defines a new sealing type that the loader can use to seal static objects.
+This macro also evaluates to the capability that permits unsealing.
+
+Static sealing capabilities are defined by both their name and the name of the compartment that exports them and so the name that you pick does not need to be unique (though, for documentation purposes, it should not be too generic).
+
+### Creating a sealed value
+
+Static sealed values are created with the `DECLARE_STATIC_SEALED_VALUE` and `DEFINE_STATIC_SEALED_VALUE` macros.
+These construct an object with the specified type and contents.
+
+**Note:** These objects may not contain pointers.
```

Comment:
What goes wrong if I try?  Is the explosion at compile time, link time, load time, or run time?  If run time, do I just see a 0 tag on the pointer value?

Reply:
I am not 100% sure.  I believe you will get a capreloc and the loader will say 'this isn't in any compartment' and abort.  We should eventually teach the linker more about where caprelocs are allowed.